server/eventlog.go

```diff
@@ -88,13 +88,37 @@ func parsePCClientEventLog(rawEventLog []byte, pcrs *tpmpb.PCRs, loader Bootload
 	}, createGroupedError("failed to fully parse MachineState:", errors)
 }
 
-func parseCanonicalEventLog(rawCanonicalEventLog []byte, pcrs *tpmpb.PCRs) (*pb.MachineState, error) {
+func parseCanonicalEventLogPCR(rawCanonicalEventLog []byte, pcrs *tpmpb.PCRs) (*pb.MachineState, error) {
+	tpm2Alg := tpm2.Algorithm(pcrs.GetHash())
+	cryptoHash, err := tpm2Alg.Hash()
+	if err != nil {
+		return nil, err
+	}
+	return getCosStateFromCEL(rawCanonicalEventLog, cryptoHash, pcrs.GetPcrs())
+}
+
+func parseCanonicalEventLogRTMR(rawCanonicalEventLog []byte, rtmrs [][]byte) (*pb.MachineState, error) {
+	mrMap := make(map[uint32][]byte)
+	mrHash := crypto.SHA384
+
+	// RTMR currently only supports SHA384, double check the hash digest size
+	for i, rtmr := range rtmrs {
+		if len(rtmr) != mrHash.Size() {
+			return nil, fmt.Errorf("got unexpected hash size for RTMR %d, want %d", len(rtmr), mrHash.Size())
+		}
+		// 0 = MRTD, 1 = RTMR[0]
+		mrMap[uint32(i)+1] = rtmr
```

Comment:
if you use register.RTMR, you should just call `Idx()`

Reply:
done